src/cargo/ops/cargo_install.rs

```diff
@@ -360,10 +361,33 @@ impl<'cfg, 'a> InstallablePackage<'cfg, 'a> {
             //
             // Note that we know at this point that _if_ bins or examples is set to `::Just`,
             // they're `::Just([])`, which is `FilterRule::none()`.
-            if self.pkg.targets().iter().any(|t| t.is_executable()) {
-                self.config
-                    .shell()
-                    .warn("none of the package's binaries are available for install using the selected features")?;
+            let binaries: Vec<_> = self.pkg.targets().iter().filter(|t| t.is_bin()).collect();
```

Comment:
Ahh good, I planned to ask about this, but forgot about it.

I changed it to `is_bin`,  because I figured that examples should not be installed. I have since learned that Cargo indeed does support installation of examples.

If I change it back to `is_executable` two questions arise:

1) 
In some of the the tests we have examples and binaries named the same, and having the message mention both as just "Target" looks weird. I guess I should try to make the message look like 
```
  Target `dummy4` requires the features: `feature4`, `feature1
  Example  `dummy4` requires the features: `feature4`, `feature1
```
Or do you have other suggestions?

2)
If I make a package with some binaries all requiring features to be enabled and an example that requires no features, I still end up in this part of the code and gets a message saying e.g.:
```
warning: none of the package's binaries are available for install using the selected features
  Target `dummy1` requires the features: `feature1`, `feature2`, `feature3`
  Target `dummy2` requires the features: `feature2`
  Target `dummy3` requires the features: `feature3`
  Target `dummy4` requires the features: `feature4`, `feature1`
  Target `dummy5` requires the features: `feature1`, `feature2`, `feature3`, `feature4`, `feature5`
Consider enabling some of them by passing, e.g., `--features="feature1 feature2 feature3"`
```
This makes me wonder if using `is_bin` is actually correct, or if there is a bug in another part of the code, that should have handed examples differently?

Reply:
I went with limiting the number of mentioned targets and adding a line saying that there are N more. I also reverted to reporting about examples as well as "real" binaries.

Reply:
Looks great! Could you add one more test to assert that new message? Thanks!

Also it seems that we need a full stop here, no?

```
- \n{} more targets also requires features not enabled. See them in the Cargo.toml file.
+ \n{} more targets also requires features not enabled. See them in the Cargo.toml file.
```

Reply:
Sure thing :-) Added a test and a full stop.